installer/pkg/config-generator/generator.go

```diff
@@ -95,6 +101,105 @@ func (c *ConfigGenerator) TectonicSystem() (string, error) {
 	})
 }
 
+// InstallConfig returns, if successful, a yaml string for on-disk install-config from 4.0 installer.
+func (c *ConfigGenerator) InstallConfig() (string, error) {
+	ic, err := c.installConfig()
+	if err != nil {
+		return "", err
+	}
+	return marshalYAML(ic)
+}
+
+func (c *ConfigGenerator) installConfig() (*types.InstallConfig, error) {
+	ic := &types.InstallConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: c.Name,
+		},
+		ClusterID: c.ClusterID,
+		Admin: types.Admin{
+			Email:    c.Admin.Email,
+			Password: c.Admin.Password,
+			SSHKey:   c.Admin.SSHKey,
+		},
+		BaseDomain: c.BaseDomain,
+		PullSecret: c.PullSecret,
+	}
+
+	ic.Networking.Type = types.NetworkType(string(c.Networking.Type))
```

Comment:
nit: maybe move this into the `ic` literal above?  Nothing in this line strikes me as something that couldn't happen there (e.g. no returned errors or other conditionals to deal with).

You could also move the `podCIDR`, `serviceCIDR`, and other setup to the top of this function, and then have a single:

```go
return &types.InstallConfig{
  ...
}, nil
```

at the end if they all succeed.

Reply:
done